### PR TITLE
simplified placeholder management

### PR DESCRIPTION
**Pull request by BigRedT**
`self._placeholders` now stores lists for list placeholders and dicts for sparse placeholders

### CODE REVIEW

**Comment by BigRedT** [comment]
i didn't add it to the previous pull request on placeholder management because of tensorflow version update.

File: aiutils/tftools/placeholder_management.py [hunk 1]
```diff
@@ -34,165 +30,185 @@ class PlaceholderManager():
...
-            name_indices = name + '_indices'
-            name_values = name + '_values'
-            name_shape = name + '_shape'
+            self._placeholders[name] = self.create_placeholder(
+                name,
+                dtype,
+                shape=shape,
+                sparse=sparse)
+
+    def create_placeholder(self, name, dtype, shape=None, sparse=False):
```
Comment (jrock08) [inline aiutils/tftools/placeholder_management.py]: I would recommend prepending __ to this to make it private.

http://www.diveintopython.net/object_oriented_framework/private_functions.html

File: aiutils/tftools/placeholder_management.py [hunk 2]
```diff
@@ -34,165 +30,185 @@ class PlaceholderManager():
...
-    def add_placeholder(self, name, dtype, shape=None, sparse=False):
+        self.issparse = set()
+        self.islist = set()
+
+    def add_placeholder(
+            self,
+            name,
+            dtype,
+            shape=None,
+            size=None,
```
Comment (jrock08) [inline aiutils/tftools/placeholder_management.py]: Not sure size is the right name.  I didn't realize it was keying to list placeholders.

Perhaps list_len=None?

File: aiutils/tftools/placeholder_management.py [hunk 3]
```diff
@@ -34,165 +30,185 @@ class PlaceholderManager():
...
+            else:
+                self.get_feed_dict_inner(
+                    feed_dict, 
+                    name,
+                    self._placeholders[name], 
+                    input_value)
 
         return feed_dict
 
+    def get_feed_dict_inner(self, feed_dict, name, plh, input_value):
```
Comment (jrock08) [inline aiutils/tftools/placeholder_management.py]: `__get_feed_dict_inner`

File: tests/test_tftools.py [hunk 4]
```diff
@@ -277,11 +277,11 @@ def test_placeholder_management():
     plh_mgr.add_placeholder('weights', tf.float64, sparse=True)
 
     # Get a dictionary of placeholders
-    plhs = plh_mgr.get_placeholders()
```
Comment (jrock08) [inline tests/test_tftools.py]: Can you add a list placeholder and a list sparse placeholder to verify that it can make them and feed to them.

**Comment by BigRedT** [comment]
@jrock08 - I have merged this branch into master after addressing all your comments. Feel free to raise an issue if you need any more changes.

**Pull request merged**